joss/paper.md

```diff
@@ -0,0 +1,103 @@
+text. It can recognize a wide variety of citations commonly appearing in American
+legal decisions, including:
+
+- full case (e.g., `Bush v. Gore, 531 U.S. 98, 99-100 (2000)`)
+- short case (e.g., `531 U.S., at 99`)
+- statutory (e.g., `Mass. Gen. Laws ch. 1, § 2`)
+- law journal (e.g., `1 Minn. L. Rev. 1`)
+- supra (e.g., `Bush, supra, at 100`)
+- id (e.g., `Id., at 101`)
+
+It also offers tools for efficiently aggregating like citations and annotating
+citations with custom markup.
+
+# Statement of need
+
+Citations are the bedrock of legal writing and a frequent topic of legal
+research, but few open-source tools exist for extracting them from legal
+texts. Because of this, researchers have historically relied on proprietary
+citation data provided by vendors like LexisNexis and Westlaw [e.g., @Black2013;
+@Fowler2007; @SpriggsII2000] or have used their own personal scripts to parse
+such data from texts ad hoc [e.g., @Clark2012; @Fowler2008]. By providing an
+open-source, standardized alternative to these approaches, `eyecite` promises
+to increase scholarly transparency and consistency. It also promises to give
+researchers the extendability and flexibility to develop new methods of
+citation analysis that are currently not possible under the prevailing
+approaches.
+
+For example, one burgeoning research agenda seeks to apply machine learning
+techniques to citation analysis, either to recommend relevant authorities to
+legal practitioners [@HoForthcoming], model the topography of the legal search
+space [@Dadgostari2021; @Leibon2018], or automatically detect and label
+the semantic purpose of citations in texts [@Sadeghian2018]. One obvious
+application of `eyecite` would be to use it to generate empirical training
+data for these kinds of machine learning tasks.
+
+To facilitate those kinds of projects and more, `eyecite` exposes significant
+entity metadata to the user. For case citations, `eyecite` parses and
+exposes information regarding a citation's textual position, year, normalized
+reporter, normalized court, volume, page, pincite page, and accompanying
+parenthetical text, as well as `eyecite`'s best guess at the names of the
+plaintiff and defendant of the cited case.  For statutory citations, `eyecite`
+parses and exposes information regarding a citation's textual position, year,
+normalized reporter, chapter, section, publisher, and accompanying parenthetical
+text. It does this using a
+[database](https://github.com/freelawproject/reporters-db) of thousands of
+regular expressions that have been accumulated and refined over the course of
+years of research.
+
+Recognizing that in addition to obtaining comprehensive results, researchers
+often want to parse many documents and citations at once, `eyecite` is also
+designed with an eye toward performance: it makes use of the Hyperscan library
+[@Wang2019] to tokenize and parse its input text in a highly efficient fashion.
```

Comment:
Worth providing a ballpark figure? I think I get on the order of 10MB/second of typical legal text through `get_citations`, though it would depend on the citation density. 

The interesting thing to highlight here is that hand-tuning is reasonable, because citations are reasonably consistent but style guides are idiosyncratic through hundreds of years of use; and hyperscan makes it possible to make that hand-tuning performant.

Reply:
I added a note about your first comment, but I don't totally get the second one. Can you propose some language to include?

Reply:
Maybe something like "... and the LexisNexis and Westlaw databases. Hyperscan, which was originally intended for security scanning network traffic against large regular expression blocklists, allows `eyecite` to apply thousands of tuned regular expressions to match the idiosyncratic ways that courts have cited each other over centuries of caselaw without a loss of performance. Additionally ..."

This could be supported by a sentence in the first paragraph like "... to parse
such data from texts ad hoc [e.g., @Clark2012; @Fowler2008]. While citations often follow a predictable format, human authors have used a wide variety of citation formats and shorthands over centuries of caselaw -- and continue to add new ones -- so accurate citation extraction requires maintenance of a long list [reporters-db] of rules and exceptions."

I'm not married to any of this -- just thinking it might be helpful to spell out why we need more than `\d+ \w+ \d+` and how cool it is that this security scanning tool all of a sudden becomes a great fit.

Reply:
Great, I like it